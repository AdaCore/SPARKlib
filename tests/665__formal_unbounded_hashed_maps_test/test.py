from test_support import prove_all, sparklib_exec_test
import os

contains_manual_proof = False
os.environ["SPARKLIB_BODY_MODE"] = "On"


if __name__ == "__main__":
    prove_all(
        counterexample=False,
        sparklib=True,
        opt=["-u", "inst.ads", "test.adb", "test_resize.adb"],
    )
    sparklib_exec_test()
    sparklib_exec_test("test_resize.gpr", "./r_obj/test_resize")
