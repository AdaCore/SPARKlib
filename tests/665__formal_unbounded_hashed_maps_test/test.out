inst.ads:6:13: info: implicit aspect Always_Terminates on "Witness" has been proved, subprogram will terminate
inst.ads:6:56: info: division check proved (CVC5: 1 VC)
inst.ads:7:13: info: implicit aspect Always_Terminates on "Hash" has been proved, subprogram will terminate
inst.ads:7:66: info: division check proved (CVC5: 1 VC)
inst.ads:7:66: info: range check proved (CVC5: 1 VC)
inst.ads:8:13: info: implicit aspect Always_Terminates on "Eq" has been proved, subprogram will terminate
inst.ads:8:55: info: division check proved (CVC5: 1 VC)
inst.ads:8:70: info: division check proved (CVC5: 1 VC)
inst.ads:9:13: info: implicit aspect Always_Terminates on "Equivalent" has been proved, subprogram will terminate
test.adb:26:29: medium: postcondition might fail, cannot prove K in 1 .. 5
test.adb:28:29: info: precondition proved (CVC5: 1 VC)
test.adb:31:23: info: unrolling loop
test.adb:32:10: medium: precondition might fail, cannot prove not Contains (Container, Key)
test.adb:38:29: medium: postcondition might fail, cannot prove K in 100 | 200 | 300 | 400 | 500
test.adb:40:29: info: precondition proved (CVC5: 1 VC)
test.adb:43:23: info: unrolling loop
test.adb:44:10: medium: precondition might fail, cannot prove not Contains (Container, Key)
test.adb:51:07: info: precondition proved (CVC5: 1 VC)
test.adb:55:07: info: initialization of "X" proved
test.adb:57:07: info: precondition proved (CVC5: 1 VC)
test.adb:59:07: info: precondition proved (CVC5: 1 VC)
test.adb:63:07: info: initialization of "X" proved
test.adb:63:10: info: initialization of "Y" proved
test.adb:63:13: info: initialization of "Z" proved
test.adb:65:07: info: precondition proved (CVC5: 1 VC)
test.adb:68:07: info: precondition proved (CVC5: 1 VC)
test.adb:70:07: info: precondition proved (CVC5: 2 VC)
test.adb:71:07: info: precondition proved (Z3: 1 VC)
test.adb:74:07: info: precondition proved (CVC5: 1 VC)
test.adb:75:07: info: precondition proved (CVC5: 1 VC)
test.adb:78:07: medium: precondition might fail, cannot prove not Contains (Container, Key)
test.adb:79:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:80:07: medium: precondition might fail
test.adb:84:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:85:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:86:07: medium: precondition might fail
test.adb:90:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:91:07: medium: precondition might fail, cannot prove not Contains (Container, Key)
test.adb:92:07: info: precondition proved (Z3: 1 VC)
test.adb:96:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:97:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:98:07: info: precondition proved (Z3: 1 VC)
test.adb:102:07: info: initialization of "X" proved
test.adb:104:07: info: precondition proved (CVC5: 1 VC)
test.adb:106:07: info: precondition proved (CVC5: 1 VC)
test.adb:110:07: info: initialization of "X" proved
test.adb:113:07: info: precondition proved (CVC5: 1 VC)
test.adb:116:07: info: precondition proved (CVC5: 1 VC)
test.adb:120:07: info: initialization of "X" proved
test.adb:120:10: info: initialization of "Y" proved
test.adb:124:07: info: precondition proved (CVC5: 1 VC)
test.adb:126:23: info: unrolling loop
test.adb:127:10: info: precondition proved (CVC5: 5 VC; Z3: 1 VC)
test.adb:130:07: info: precondition proved (CVC5: 1 VC)
test.adb:134:07: info: initialization of "X" proved
test.adb:139:10: info: precondition proved (CVC5: 1 VC)
test.adb:145:10: info: precondition proved (CVC5: 1 VC)
test.adb:150:07: info: initialization of "X" proved
test.adb:153:07: info: precondition proved (CVC5: 1 VC)
test.adb:153:15: info: precondition proved (CVC5: 1 VC)
test.adb:158:07: info: initialization of "X" proved
test.adb:161:07: info: precondition proved (CVC5: 1 VC)
test.adb:161:15: info: precondition proved (CVC5: 1 VC)
test.adb:161:49: info: precondition proved (CVC5: 1 VC)
test.adb:166:07: info: initialization of "X" proved
test.adb:169:07: info: precondition proved (CVC5: 1 VC)
test.adb:169:15: info: precondition proved (CVC5: 1 VC)
test.adb:173:07: info: initialization of "X" proved
test.adb:174:07: info: initialization of "P1" proved
test.adb:174:11: info: initialization of "P2" proved
test.adb:175:07: info: initialization of "E1" proved
test.adb:176:07: info: initialization of "K1" proved
test.adb:176:11: info: initialization of "K2" proved
test.adb:180:13: info: precondition proved (CVC5: 1 VC)
test.adb:181:13: info: precondition proved (CVC5: 1 VC)
test.adb:182:13: info: precondition proved (CVC5: 1 VC)
test.adb:183:13: info: precondition proved (CVC5: 1 VC)
test.adb:185:07: info: precondition proved (CVC5: 1 VC)
test.adb:187:07: info: precondition proved (CVC5: 1 VC)
test.adb:187:15: info: precondition proved (CVC5: 1 VC)
test.adb:188:07: info: precondition proved (CVC5: 1 VC)
test.adb:189:07: info: precondition proved (CVC5: 1 VC)
test.adb:189:15: info: precondition proved (CVC5: 1 VC)
test.adb:189:36: info: precondition proved (CVC5: 1 VC)
test.adb:190:07: info: precondition proved (Z3: 1 VC)
test.adb:190:15: info: precondition proved (CVC5: 1 VC)
test.adb:195:07: info: initialization of "X" proved
test.adb:196:07: info: initialization of "P1" proved
test.adb:197:07: info: initialization of "E1" proved
test.adb:201:13: info: precondition proved (CVC5: 1 VC)
test.adb:203:07: info: precondition proved (CVC5: 1 VC)
test.adb:203:15: info: precondition proved (CVC5: 1 VC)
test.adb:208:07: info: initialization of "X" proved
test.adb:209:07: warning: unused variable "P1"
test.adb:213:07: info: precondition proved (CVC5: 1 VC)
test.adb:213:15: info: precondition proved (CVC5: 1 VC)
test.adb:218:07: info: initialization of "X" proved
test.adb:219:07: info: initialization of "P2" proved
test.adb:220:07: info: initialization of "P4" proved
test.adb:221:07: info: initialization of "K2" proved
test.adb:221:11: info: initialization of "K4" proved
test.adb:222:07: info: initialization of "E2" proved
test.adb:222:11: info: initialization of "E4" proved
test.adb:225:13: info: precondition proved (CVC5: 1 VC)
test.adb:226:13: info: precondition proved (CVC5: 1 VC)
test.adb:227:13: info: precondition proved (CVC5: 1 VC)
test.adb:228:13: info: precondition proved (CVC5: 1 VC)
test.adb:228:22: info: precondition proved (CVC5: 1 VC)
test.adb:229:13: info: precondition proved (CVC5: 1 VC)
test.adb:230:13: info: precondition proved (CVC5: 1 VC)
test.adb:234:36: info: precondition proved (CVC5: 1 VC)
test.adb:234:47: info: null exclusion check proved (CVC5: 1 VC)
test.adb:236:10: info: precondition proved (CVC5: 1 VC)
test.adb:236:20: info: pointer dereference check proved (CVC5: 1 VC)
test.adb:237:12: info: pointer dereference check proved (CVC5: 1 VC)
test.adb:242:07: info: precondition proved (Z3: 1 VC)
test.adb:242:44: info: precondition proved (CVC5: 1 VC)
test.adb:243:07: info: precondition proved (Z3: 1 VC)
test.adb:243:15: info: precondition proved (Z3: 1 VC)
test.adb:244:07: info: precondition proved (Z3: 1 VC)
test.adb:245:07: info: precondition proved (Z3: 1 VC)
test.adb:245:44: info: precondition proved (CVC5: 1 VC)
test.adb:245:70: info: precondition proved (CVC5: 1 VC)
test.adb:246:07: info: precondition proved (Z3: 1 VC)
test.adb:246:20: info: precondition proved (CVC5: 1 VC)
test.adb:246:49: info: precondition proved (CVC5: 1 VC)
test.adb:246:58: info: precondition proved (Z3: 1 VC)
test.adb:251:07: info: initialization of "X" proved
test.adb:257:36: info: precondition proved (CVC5: 1 VC)
test.adb:257:47: info: null exclusion check proved (CVC5: 1 VC)
test.adb:259:10: medium: precondition might fail
test.adb:259:20: info: pointer dereference check proved (CVC5: 1 VC)
test.adb:260:12: info: pointer dereference check proved (CVC5: 1 VC)
test.adb:265:07: info: precondition proved (CVC5: 1 VC)
test.adb:265:15: info: precondition proved (CVC5: 1 VC)
test.adb:266:07: info: precondition proved (CVC5: 1 VC)
test.adb:267:07: medium: precondition might fail
test.adb:267:15: info: precondition proved (Z3: 1 VC)
test.adb:271:07: info: initialization of "X" proved
test.adb:271:10: info: initialization of "Y" proved
test.adb:275:07: info: precondition proved (CVC5: 1 VC)
test.adb:277:23: info: unrolling loop
test.adb:278:10: info: precondition proved (CVC5: 5 VC; Z3: 1 VC)
test.adb:281:07: info: precondition proved (CVC5: 1 VC)
test.adb:286:07: info: initialization of "X" proved
test.adb:286:10: info: initialization of "Y" proved
test.adb:287:07: info: initialization of "P1" proved
test.adb:287:11: info: initialization of "P11" proved
test.adb:287:16: info: initialization of "P21" proved
test.adb:287:21: info: initialization of "P31" proved
test.adb:287:26: info: initialization of "P2" proved
test.adb:288:07: info: initialization of "B1" proved
test.adb:288:11: info: initialization of "B2" proved
test.adb:288:15: info: initialization of "B3" proved
test.adb:290:07: info: precondition proved (CVC5: 1 VC)
test.adb:291:07: info: precondition proved (CVC5: 1 VC)
test.adb:292:07: info: precondition proved (CVC5: 1 VC)
test.adb:293:07: info: precondition proved (CVC5: 1 VC)
test.adb:293:15: info: precondition proved (CVC5: 1 VC)
test.adb:294:07: info: precondition proved (Z3: 1 VC)
test.adb:294:15: info: precondition proved (CVC5: 1 VC)
test.adb:297:07: info: precondition proved (CVC5: 1 VC)
test.adb:298:07: info: precondition proved (CVC5: 1 VC)
test.adb:299:07: info: precondition proved (CVC5: 1 VC)
test.adb:300:07: medium: precondition might fail
test.adb:301:07: medium: precondition might fail
test.adb:302:07: medium: precondition might fail
test.adb:302:15: medium: precondition might fail
test.adb:302:38: medium: precondition might fail
test.adb:302:61: info: precondition proved (CVC5: 1 VC)
test.adb:303:07: medium: precondition might fail
test.adb:303:15: info: precondition proved (Z3: 1 VC)
test.adb:303:42: info: precondition proved (Z3: 1 VC)
test.adb:303:69: info: precondition proved (CVC5: 1 VC)
test.adb:305:07: info: precondition proved (Z3: 1 VC)
test.adb:306:07: medium: precondition might fail [possible fix: postcondition of call at line 305 should mention Inserted (for argument B1)]
test.adb:307:07: medium: precondition might fail
test.adb:308:07: medium: precondition might fail
test.adb:308:15: info: precondition proved (CVC5: 1 VC)
test.adb:309:07: medium: precondition might fail
test.adb:309:15: info: precondition proved (CVC5: 1 VC)
test.adb:310:07: info: precondition proved (Z3: 1 VC)
test.adb:310:15: info: precondition proved (Z3: 1 VC)
test.adb:311:07: medium: precondition might fail
test.adb:311:15: medium: precondition might fail
test.adb:313:07: info: precondition proved (Z3: 1 VC)
test.adb:314:07: info: precondition proved (Z3: 1 VC)
test.adb:315:07: info: precondition proved (Z3: 1 VC)
test.adb:316:07: medium: precondition might fail
test.adb:317:07: medium: precondition might fail
test.adb:317:15: info: precondition proved (CVC5: 1 VC)
test.adb:317:35: info: precondition proved (CVC5: 1 VC)
test.adb:322:07: info: initialization of "X" proved
test.adb:322:10: info: initialization of "Y" proved
test.adb:323:07: info: initialization of "P21" proved
test.adb:323:12: info: initialization of "P2" proved
test.adb:325:07: info: precondition proved (CVC5: 2 VC)
test.adb:326:07: info: precondition proved (CVC5: 1 VC)
test.adb:327:07: info: precondition proved (CVC5: 1 VC)
test.adb:328:07: info: precondition proved (CVC5: 1 VC)
test.adb:328:15: info: precondition proved (CVC5: 1 VC)
test.adb:330:07: info: precondition proved (CVC5: 2 VC)
test.adb:331:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:332:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:333:07: info: precondition proved (CVC5: 1 VC)
test.adb:334:07: info: precondition proved (Z3: 1 VC)
test.adb:335:07: info: precondition proved (Z3: 1 VC)
test.adb:335:15: info: precondition proved (Z3: 1 VC)
test.adb:335:42: info: precondition proved (Z3: 1 VC)
test.adb:335:69: info: precondition proved (CVC5: 1 VC)
test.adb:337:07: info: precondition proved (CVC5: 1 VC; Z3: 1 VC)
test.adb:338:07: medium: precondition might fail, cannot prove not Contains (Container, Key)
test.adb:339:07: info: precondition proved (CVC5: 1 VC)
test.adb:340:07: info: precondition proved (CVC5: 1 VC)
test.adb:341:07: info: precondition proved (CVC5: 1 VC)
test.adb:341:15: info: precondition proved (CVC5: 1 VC)
test.adb:342:07: info: precondition proved (Z3: 1 VC)
test.adb:342:15: info: precondition proved (Z3: 1 VC)
test.adb:346:07: medium: precondition might fail
test.adb:346:15: info: precondition proved (Z3: 1 VC)
test.adb:346:35: info: precondition proved (Z3: 1 VC)
test.adb:350:07: info: initialization of "X" proved
test.adb:350:10: info: initialization of "Y" proved
test.adb:351:07: info: initialization of "P21" proved
test.adb:351:12: info: initialization of "P2" proved
test.adb:353:07: info: precondition proved (CVC5: 1 VC)
test.adb:354:07: info: precondition proved (CVC5: 1 VC)
test.adb:355:07: info: precondition proved (CVC5: 1 VC)
test.adb:356:07: info: precondition proved (CVC5: 1 VC)
test.adb:356:15: info: precondition proved (CVC5: 1 VC)
test.adb:358:07: info: precondition proved (CVC5: 1 VC)
test.adb:359:07: info: precondition proved (CVC5: 1 VC)
test.adb:360:07: info: precondition proved (CVC5: 1 VC)
test.adb:361:07: medium: precondition might fail
test.adb:362:07: info: precondition proved (Z3: 1 VC)
test.adb:363:07: medium: precondition might fail
test.adb:363:15: info: precondition proved (Z3: 1 VC)
test.adb:363:42: info: precondition proved (Z3: 1 VC)
test.adb:363:69: info: precondition proved (CVC5: 1 VC)
test.adb:365:07: info: precondition proved (CVC5: 1 VC)
test.adb:366:07: info: precondition proved (CVC5: 1 VC)
test.adb:367:07: medium: precondition might fail
test.adb:368:07: info: precondition proved (CVC5: 1 VC)
test.adb:369:07: info: precondition proved (CVC5: 1 VC)
test.adb:369:15: info: precondition proved (CVC5: 1 VC)
test.adb:370:07: info: precondition proved (Z3: 1 VC)
test.adb:374:07: medium: precondition might fail
test.adb:374:15: info: precondition proved (Z3: 1 VC)
test.adb:374:35: info: precondition proved (Z3: 1 VC)
test.adb:376:07: info: precondition proved (Z3: 1 VC)
test.adb:377:07: info: precondition proved (Z3: 1 VC)
test.adb:378:07: medium: precondition might fail
test.adb:379:07: info: precondition proved (Z3: 1 VC)
test.adb:379:15: info: precondition proved (CVC5: 1 VC)
test.adb:379:44: info: precondition proved (CVC5: 1 VC)
test.adb:380:07: medium: precondition might fail
test.adb:380:15: info: precondition proved (Z3: 1 VC)
test.adb:380:38: info: precondition proved (Z3: 1 VC)
test.adb:384:07: info: initialization of "X" proved
test.adb:384:10: info: initialization of "Y" proved
test.adb:385:07: info: initialization of "P" proved
test.adb:385:10: info: initialization of "Q" proved
test.adb:390:07: info: precondition proved (CVC5: 1 VC)
test.adb:392:07: info: precondition proved (CVC5: 1 VC)
test.adb:393:07: info: precondition proved (Z3: 1 VC)
test.adb:393:15: info: precondition proved (CVC5: 1 VC)
test.adb:394:07: info: precondition proved (CVC5: 1 VC)
test.adb:394:15: info: precondition proved (Z3: 1 VC)
test.adb:395:07: info: precondition proved (Z3: 1 VC)
test.adb:395:15: info: precondition proved (CVC5: 1 VC)
test.adb:395:34: info: precondition proved (Z3: 1 VC)
test.adb:400:07: info: precondition proved (Z3: 1 VC)
test.adb:402:07: info: precondition proved (CVC5: 1 VC)
test.adb:403:07: medium: precondition might fail
test.adb:403:15: info: precondition proved (CVC5: 1 VC)
test.adb:404:07: info: precondition proved (CVC5: 1 VC)
test.adb:404:15: info: precondition proved (Z3: 1 VC)
test.adb:405:07: medium: precondition might fail
test.adb:405:15: info: precondition proved (Z3: 1 VC)
test.adb:405:36: info: precondition proved (Z3: 1 VC)
test.adb:409:07: info: initialization of "X" proved
test.adb:409:10: info: initialization of "Y" proved
test.adb:410:07: info: initialization of "Q" proved
test.adb:413:07: info: precondition proved (CVC5: 1 VC)
test.adb:418:07: info: precondition proved (Z3: 1 VC)
test.adb:421:22: info: assertion proved (CVC5: 1 VC)
test.adb:424:07: medium: precondition might fail
test.adb:425:07: info: precondition proved (Z3: 1 VC)
test.adb:426:07: medium: precondition might fail
test.adb:426:15: info: precondition proved (Z3: 1 VC)
test.adb:427:07: medium: precondition might fail
test.adb:427:15: info: precondition proved (Z3: 1 VC)
test.adb:432:07: info: precondition proved (Z3: 1 VC)
test.adb:435:22: info: assertion proved (Z3: 1 VC)
test.adb:438:07: medium: precondition might fail
test.adb:439:07: info: precondition proved (Z3: 1 VC)
test.adb:440:07: medium: precondition might fail
test.adb:440:15: info: precondition proved (Z3: 1 VC)
test.adb:441:07: medium: precondition might fail
test.adb:441:15: info: precondition proved (Z3: 1 VC)
test.adb:446:07: info: initialization of "X" proved
test.adb:446:10: info: initialization of "Y" proved
test.adb:447:07: info: initialization of "Q" proved
test.adb:451:07: medium: precondition might fail
test.adb:453:07: info: precondition proved (CVC5: 1 VC)
test.adb:454:07: info: precondition proved (CVC5: 1 VC)
test.adb:455:07: medium: precondition might fail
test.adb:455:15: info: precondition proved (Z3: 1 VC)
test.adb:456:07: info: precondition proved (Z3: 1 VC)
test.adb:456:15: info: precondition proved (Z3: 1 VC)
test.adb:460:07: info: precondition proved (Z3: 1 VC)
test.adb:462:07: info: precondition proved (CVC5: 1 VC)
test.adb:463:07: info: precondition proved (Z3: 1 VC)
test.adb:464:07: medium: precondition might fail
test.adb:464:15: info: precondition proved (Z3: 1 VC)
test.adb:465:07: info: precondition proved (Z3: 1 VC)
test.adb:465:15: info: precondition proved (Z3: 1 VC)
test.adb:470:07: info: initialization of "X" proved
test.adb:470:10: info: initialization of "Y" proved
test.adb:471:07: info: initialization of "P" proved
test.adb:471:10: info: initialization of "Q" proved
test.adb:476:07: medium: precondition might fail
test.adb:478:07: info: precondition proved (CVC5: 1 VC)
test.adb:479:07: info: precondition proved (Z3: 1 VC)
test.adb:480:07: info: precondition proved (Z3: 1 VC)
test.adb:480:15: info: precondition proved (Z3: 1 VC)
test.adb:481:07: medium: precondition might fail
test.adb:481:15: info: precondition proved (Z3: 1 VC)
test.adb:486:07: info: precondition proved (Z3: 1 VC)
test.adb:488:07: info: precondition proved (CVC5: 1 VC)
test.adb:489:07: info: precondition proved (Z3: 1 VC)
test.adb:490:07: medium: precondition might fail
test.adb:490:15: info: precondition proved (Z3: 1 VC)
test.adb:491:07: info: precondition proved (Z3: 1 VC)
test.adb:491:15: info: precondition proved (Z3: 1 VC)
test.adb:495:07: info: initialization of "X" proved
test.adb:497:07: info: precondition proved (CVC5: 1 VC)
test.adb:499:07: info: precondition proved (CVC5: 1 VC)
test.adb:499:46: info: precondition proved (CVC5: 1 VC)
test.adb:503:07: info: initialization of "X" proved
test.adb:503:10: info: initialization of "Y" proved
test.adb:508:07: info: precondition proved (CVC5: 1 VC)
test.adb:508:15: info: precondition proved (CVC5: 1 VC)
test.adb:509:07: info: precondition proved (CVC5: 1 VC)
test.adb:510:07: info: precondition proved (CVC5: 1 VC)
test.adb:513:07: info: precondition proved (CVC5: 1 VC)
test.adb:513:31: info: precondition proved (CVC5: 1 VC)
test.adb:514:07: info: precondition proved (CVC5: 1 VC)
test.adb:515:07: info: precondition proved (CVC5: 1 VC)
test.adb:520:07: info: precondition proved (CVC5: 1 VC)
test.adb:520:31: info: precondition proved (CVC5: 1 VC)
test.adb:521:07: info: precondition proved (CVC5: 1 VC)
test.adb:522:07: info: precondition proved (Z3: 1 VC)
test.adb:526:07: info: initialization of "X" proved
test.adb:526:10: info: initialization of "Y" proved
test.adb:528:07: info: precondition proved (CVC5: 1 VC)
test.adb:532:07: info: precondition proved (CVC5: 1 VC)
test.adb:533:07: info: precondition proved (Z3: 1 VC)
test.adb:534:07: medium: precondition might fail
test.adb:538:07: info: precondition proved (Z3: 1 VC)
test.adb:539:07: info: precondition proved (Z3: 1 VC)
test.adb:540:07: info: precondition proved (Z3: 1 VC)
test.adb:544:07: info: initialization of "X" proved
test.adb:544:10: info: initialization of "Y" proved
test.adb:546:07: info: precondition proved (CVC5: 1 VC)
test.adb:550:07: info: precondition proved (CVC5: 1 VC)
test.adb:551:07: info: precondition proved (CVC5: 1 VC)
test.adb:552:07: medium: precondition might fail
test.adb:556:07: info: precondition proved (CVC5: 1 VC)
test.adb:557:07: info: precondition proved (Z3: 1 VC)
test.adb:558:07: info: precondition proved (Z3: 1 VC)
test.adb:562:07: info: initialization of "X" proved
test.adb:563:07: info: initialization of "P" proved
test.adb:566:07: info: precondition proved (CVC5: 1 VC)
test.adb:568:07: info: precondition proved (CVC5: 1 VC)
test.adb:569:07: info: precondition proved (CVC5: 1 VC)
test.adb:570:07: info: precondition proved (CVC5: 1 VC)
test.adb:574:07: info: initialization of "X" proved
test.adb:574:10: info: initialization of "Y" proved
test.adb:576:07: info: precondition proved (CVC5: 1 VC)
test.adb:576:34: info: precondition proved (CVC5: 1 VC)
test.adb:577:07: info: precondition proved (CVC5: 1 VC)
test.adb:580:07: info: precondition proved (CVC5: 1 VC)
test.adb:580:34: info: precondition proved (CVC5: 1 VC)
test.adb:581:07: medium: precondition might fail
test.adb:581:35: info: precondition proved (CVC5: 1 VC)
test.adb:584:07: info: precondition proved (CVC5: 1 VC)
test.adb:584:34: info: precondition proved (CVC5: 1 VC)
test.adb:585:07: info: precondition proved (Z3: 1 VC)
test.adb:585:35: info: precondition proved (CVC5: 1 VC)
test.adb:589:18: info: keys shall be distinct and all elements shall be smaller than "Aggr_Capacity" for maps aggregates at spark-containers-formal-unbounded_hashed_maps.ads:81, in instantiation at inst.ads:10
test.adb:589:18: info: precondition proved (CVC5: 11 VC)
test.adb:591:07: info: precondition proved (CVC5: 1 VC)
test.adb:592:07: info: precondition proved (CVC5: 1 VC)
test.adb:593:07: info: precondition proved (CVC5: 1 VC)
test.adb:593:15: info: precondition proved (CVC5: 1 VC)
test.adb:593:38: info: precondition proved (CVC5: 1 VC)
test.adb:597:04: info: precondition proved (Trivial: 1 VC)
test.adb:598:04: info: precondition proved (Trivial: 1 VC)
test.adb:599:04: info: precondition proved (Trivial: 1 VC)
test.adb:600:04: info: precondition proved (Trivial: 1 VC)
test.adb:601:04: info: precondition proved (Trivial: 1 VC)
test.adb:602:04: info: precondition proved (Trivial: 1 VC)
test.adb:603:04: info: precondition proved (Trivial: 1 VC)
test.adb:604:04: info: precondition proved (Trivial: 1 VC)
test.adb:605:04: info: precondition proved (Trivial: 1 VC)
test.adb:606:04: info: precondition proved (Trivial: 1 VC)
test.adb:607:04: info: precondition proved (Trivial: 1 VC)
test.adb:608:04: info: precondition proved (Trivial: 1 VC)
test.adb:609:04: info: precondition proved (Trivial: 1 VC)
test.adb:610:04: info: precondition proved (Trivial: 1 VC)
test.adb:611:04: info: precondition proved (Trivial: 1 VC)
test.adb:612:04: info: precondition proved (Trivial: 1 VC)
test.adb:613:04: info: precondition proved (Trivial: 1 VC)
test.adb:614:04: info: precondition proved (Trivial: 1 VC)
test.adb:615:04: info: precondition proved (Trivial: 1 VC)
test.adb:616:04: info: precondition proved (Trivial: 1 VC)
test.adb:617:04: info: precondition proved (Trivial: 1 VC)
test.adb:618:04: info: precondition proved (Trivial: 1 VC)
test.adb:619:04: info: precondition proved (Trivial: 1 VC)
test.adb:620:04: info: precondition proved (Trivial: 1 VC)
test.adb:621:04: info: precondition proved (Trivial: 1 VC)
test.adb:622:04: info: precondition proved (Trivial: 1 VC)
test.adb:623:04: info: precondition proved (Trivial: 1 VC)
test.adb:624:04: info: precondition proved (Trivial: 1 VC)
test.adb:625:04: info: precondition proved (Trivial: 1 VC)
test.adb:626:04: info: precondition proved (Trivial: 1 VC)
test_resize.adb:10:04: info: initialization of "X" proved
test_resize.adb:13:07: info: precondition proved (CVC5: 4 VC)
test_resize.adb:14:30: info: loop invariant initialization proved (CVC5: 1 VC)
test_resize.adb:14:30: info: loop invariant preservation proved (CVC5: 1 VC)
test_resize.adb:15:30: info: loop invariant initialization proved (CVC5: 2 VC)
test_resize.adb:15:48: medium: loop invariant might not be preserved by an arbitrary iteration, cannot prove K in 1 .. I
inst.ads:10:04: warning: check will fail at run time [-gnatw.a]
inst.ads:10:04: warning: check will fail at run time [-gnatw.a]
inst.ads:10:04: warning: in instantiation at spark-containers-formal-unbounded_hashed_maps.ads:236 [-gnatw.a]
inst.ads:10:04: warning: in instantiation at spark-containers-formal-unbounded_hashed_maps.ads:236 [-gnatw.a]
inst.ads:10:04: warning: in instantiation at spark-containers-functional-maps.ads:134 [-gnatw.a]
inst.ads:10:04: warning: in instantiation at spark-containers-functional-maps.ads:153 [-gnatw.a]
